slick/src/main/scala/slick/dbio/DBIOAction.scala

```diff
@@ -114,6 +114,22 @@ sealed trait DBIOAction[+R, +S <: NoStream, -E <: Effect] extends Dumpable {
   def collect[R2](pf: PartialFunction[R,R2])(implicit executor: ExecutionContext): DBIOAction[R2, NoStream, E] =
     map(r1 => pf.applyOrElse(r1,(r:R) => throw new NoSuchElementException(s"DBIOAction.collect partial function is not defined at: $r")))
 
+  /** Replace a result of a successful execution of this action with the unit value.
+    * If this action fails, the resulting action also fails.
+    *
+    * A shortcut for `.map(_ => ())`.
+    */
+  def void(implicit executor: ExecutionContext): DBIOAction[Unit, NoStream, E] =
+    map(_ => ())
```

Comment:
I think we can avoid taking an ExecutionContext by using the parasitic execution context.

Reply:
The `ExecutionContext.parasitic`, right? Or do you mean `DBIO.sameThreadExecutionContext`?

Reply:
That's a great question. Is there any meaningful difference?

If yes and there's any reason to use the former, one question would be if it exists in scala 2.12, although we really should drop it IMO now that 3.5.0 is finally out.

OTOH if there's either no difference, or `parasitic` is better and available to us, the question is whether `DBIO.sameThreadExecutionContext` should continue to exist altogether. If not maybe it could be made to alias `parasitic` and deprecated. Also if everywhere else that's what's used then there's the consistency factor.

:shrug: 

I don't know the answer to any of that. If you could research it would be great.

One possible strategy would be to use `DBIO.sameThreadExecutionContext?` for now in this PR, and in a separate PR make it a deprecated alias...

Reply:
According to the code, the logic is similar to `DBIO.sameThreadExecutionContext`:
https://github.com/scala/scala/blob/2.13.x/src/library/scala/concurrent/ExecutionContext.scala#L221-L225
https://github.com/scala/scala/blob/2.13.x/src/library/scala/concurrent/BatchingExecutor.scala#L245-L269

I would prefer using `DBIO.sameThreadExecutionContext` since it's already used in `flatten`.